server/boot/observe-changes.js

```diff
@@ -0,0 +1,42 @@
+var _ = require('lodash');
+var debug = require('debug')('strong-mesh-models:server:observe-changes');
+
+module.exports = function observeChanges(server) {
+  var ServiceInstance = server.models.ServiceInstance;
+  var ServiceProcess = server.models.ServiceProcess;
+  var ServerService = server.models.ServerService;
+
+  ServiceInstance.observe('after save', function(ctx, next) {
+    var inst = ctx.instance || ctx.data;
+    ServerService.__changeEmitter.emit('svc-' + inst.serverServiceId, {
+      target: 'ServiceInstance-' + inst.id,
+      type: ctx.isNewInstance ? 'create' : 'update',
+      data: _.omit(inst.toObject(), [
+        'executorId',
+        'currentDeploymentId',
+        'containerVersionInfo',
+        'PMPort',
+        'commit',
+        'token',
+        'driverMeta',
+      ]),
+      optimistic: false,
+    });
+    setImmediate(next);
+  });
+
+  ServiceProcess.observe('after save', function(ctx, next) {
+    var proc = ctx.instance || ctx.data;
+    ServiceInstance.findById(proc.serviceInstanceId, function(err, instance) {
+      if (err) return debug('Unable to locate instnce for proc %j', proc);
```

Comment:
Does the observers API not expect `next()` to be called with an error in this case?